src/test/java/com/aws/greengrass/mqttbridge/MQTTBridgeTest.java

```diff
@@ -102,25 +99,45 @@ void cleanup() {
     }
 
     private void startKernelWithConfig(String configFileName) throws InterruptedException {
+        startKernelWithConfig(configFileName, State.RUNNING);
+    }
+
+    private void startKernelWithConfig(String configFileName, State expectedStartingState) throws InterruptedException {
         CountDownLatch bridgeRunning = new CountDownLatch(1);
         kernel.parseArgs("-r", rootDir.toAbsolutePath().toString(), "-i",
                 getClass().getResource(configFileName).toString());
         listener = (GreengrassService service, State was, State newState) -> {
-            if (service.getName().equals(MQTTBridge.SERVICE_NAME) && service.getState().equals(State.RUNNING)) {
+            if (service.getName().equals(MQTTBridge.SERVICE_NAME) && service.getState().equals(expectedStartingState)) {
                 bridgeRunning.countDown();
             }
         };
         kernel.getContext().addGlobalStateChangeListener(listener);
         kernel.launch();
 
-        Assertions.assertTrue(bridgeRunning.await(TEST_TIME_OUT_SEC, TimeUnit.SECONDS));
+        assertTrue(bridgeRunning.await(TEST_TIME_OUT_SEC, TimeUnit.SECONDS));
     }
 
     @Test
     void GIVEN_Greengrass_with_mqtt_bridge_WHEN_start_kernel_THEN_bridge_starts_successfully() throws Exception {
         startKernelWithConfig("config.yaml");
     }
 
+    @Test
+    void GIVEN_Greengrass_with_mqtt_bridge_WHEN_invalid_credential_config_is_fixed_THEN_bridge_automatically_reinstalls(ExtensionContext context) throws Exception {
+        ignoreExceptionOfType(context, IllegalArgumentException.class);
+        startKernelWithConfig("config_with_invalid_credentials.yaml", State.BROKEN);
```

Comment:
Is there a way to assert the reason for the BROKEN state?

Reply:
yeah but it would be a bit of a hack to do so....`GreengrassService` doesn't expose this directly, but we could use reflection to look at `GreengrassService#error`.

Reply:
Yes you can absolutely assert on the reason. Use a log listener to listen for the error log

Reply:
ex: https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/c2e1d321b1a9cc082dcfe902ff8d03dbe55ca810/src/integrationtests/java/com/aws/greengrass/integrationtests/provisioning/ProvisionFromPluginTest.java#L100

Reply:
thanks michael, will add that in 👍